src/applications/check-in/components/layout/AppWrapper.jsx

```diff
@@ -0,0 +1,90 @@
+import React, { useEffect, useMemo, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import PropTypes from 'prop-types';
+import { useSelector, useDispatch } from 'react-redux';
+
+import { makeSelectCurrentContext, makeSelectForm } from '../../selectors';
+import { setForm } from '../../actions/universal';
+import { createSetSession } from '../../actions/authentication';
+import { useSessionStorage } from '../../hooks/useSessionStorage';
+import { useGetCheckInData } from '../../hooks/useGetCheckInData';
+import { useFormRouting } from '../../hooks/useFormRouting';
+import { isAnInternalPage } from '../../utils/navigation';
+
+const AppWrapper = props => {
+  const { children, router } = props;
+  const location = window.location.pathname;
+  const { t } = useTranslation();
+  const dispatch = useDispatch();
+  const {
+    getProgressState,
+    setProgressState,
+    getCurrentToken,
+    getPermissions,
+  } = useSessionStorage(false);
+  const selectCurrentContext = useMemo(makeSelectCurrentContext, []);
+  const selectForm = useMemo(makeSelectForm, []);
+  const currentForm = useSelector(selectForm);
+  const progressState = getProgressState(window);
+  const { checkInDataError, refreshCheckInData, isLoading } = useGetCheckInData(
+    false,
```

Comment:
NITPICK: maybe consider an object instead of a long list of boolean? 

``` 
useGetCheckinData({doAThing:true, doThatThatThing:false}) 
```

Reply:
oh yeah, I always prefer that, but forget to do it myself